plugins/kubernetesapi-report/src/main/java/com/freenow/sauron/plugins/readers/KubernetesContainersReader.java

```diff
@@ -0,0 +1,68 @@
+package com.freenow.sauron.plugins.readers;
+
+import com.freenow.sauron.model.DataSet;
+import com.freenow.sauron.plugins.commands.KubernetesGetDeploymentSpecCommand;
+import com.freenow.sauron.plugins.utils.RetryCommand;
+import com.freenow.sauron.plugins.utils.RetryConfig;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.models.*;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.freenow.sauron.plugins.utils.KubernetesResources.DEPLOYMENT;
+
+@Slf4j
+@RequiredArgsConstructor
+public class KubernetesContainersReader
+{
+    private final KubernetesGetDeploymentSpecCommand kubernetesGetDeploymentSpecCommand;
+    private final RetryConfig retryConfig;
+    public static final String PRODUCTION_READINESS = "productionReadiness";
+
+    public KubernetesContainersReader()
+    {
+        this.kubernetesGetDeploymentSpecCommand = new KubernetesGetDeploymentSpecCommand();
+        this.retryConfig = new RetryConfig();
+    }
+
+    public void read(DataSet input, String serviceLabel, ApiClient apiClient)
```

Comment:
Here we want to receive the `container property` we want to read as an argument of `read(...` method. That will be based on the plugin configuration. This way we make it dynamic and prepared for more assertions in the future.